package-res/resources/web/pentaho/lang/Event.js

```diff
@@ -118,6 +119,17 @@ define([
     },
 
     /**
+     * Gets the reason why the event was canceled.
+     *
+     * @type {!Base.Error}
+     * @readonly
+     */
+    get cancelReason() {
+      return this.isCanceled ? this._cancelReason : null;
```

Comment:
the idea here was to ensure cancelReason is either null or a proper instance of Error.